ci: add GH action and remove Travis CI integration

agnes512 opened the pull request:
Notice that we were using mongodb 3.2 for Travis. That version runs into connection errors in GitHub action. This PR uses the latest stable version of mongodb (4.4) for Github action.

<!--
Please provide a high-level description of the changes made by your pull request.

Include references to all related GitHub issues and other pull requests, for example:

Fixes #123
Implements #254
See also #23
-->

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [ ] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

dhmlau commented on .github/workflows/continous-integration.yaml:
```diff
@@ -0,0 +1,37 @@
+name: Continous Integration
```
typo

```suggestion
name: Continuous Integration
```

dhmlau commented on .github/workflows/continous-integration.yaml:
```diff
@@ -0,0 +1,37 @@
+    - run: npm test
+      env:
+        CI: true
+        
```
It's complaining no new line. Is it because there's extra space on the last line?

agnes512 replied:
Yes you're right. Fixed!

dhmlau reviewed (commented):
I have limited knowledge on GitHub Actions, but it LGTM. Please see my minor comments.

bajtos commented on .travis.yml:
```diff
@@ -1,21 +0,0 @@
-services:
-  - mongodb
-
-after_success: npm run coverage
```
I believe this command is publishing code coverage data to https://coveralls.io/github/strongloop/loopback-connector-mongodb. Can you please preserve this step in the new GHA workflow too?

I think it should be enough to add a new step at the end of the file:

```
    - run: npm install
    - run: npm test
      env:
        CI: true
    - run: npm run coverage
```

bajtos reviewed (approved):
Great job, @agnes512! I have one comment to address, PTAL 👇🏻 

Considering that our CI is broken right now, I think it's ok to land this change as-is even if it's breaking code coverage, and then restore Coveralls integration in a follow-up pull request.

agnes512 commented:
I was investigating adding coveralls to the workflow. But GitHub action seems disappeared 🤔 I am only seeing jenkins running. Not sure if I messed up something.

agnes512 commented:
@bajtos I tried to enable coveralls like https://coveralls.io/builds/32304908. But I don't know how to do it like Travis does. Seems like Travis has some repo token that coveralls.io would use. Not sure how this can be done with `nyc` + `github action`. I tried all possible solutions that I could fine but still still have no luck. 

Also, seems like the `nyc report | coveralls` still provides some coverage report:
![image](https://user-images.githubusercontent.com/50331796/106683893-5f653c00-6593-11eb-858a-386226997b08.png)

and it has `.nyc_output` locally, which contains such reports:
![image](https://user-images.githubusercontent.com/50331796/106683972-86bc0900-6593-11eb-982c-658e5d0a0c2e.png)


Not sure if it's possible to have pages like coveralls.io for our case. Would you want me to invest more? 

I also noticed that when we changed from Travis to Github action, we dropped the coveralls report for loopback-next. https://github.com/strongloop/loopback-next/pull/5110 Again, not sure how complicated it would be. But we might want to have the report for loopback-next. Please let me know if you have any concerns. Thanks.
